build/accelerate_launch.py

```diff
@@ -47,6 +48,40 @@
 ERROR_LOG = "/dev/termination-log"
 
 
+def copytree(source, destination, symlinks=False, ignore=None):
+    if not os.path.exists(destination):
+        os.makedirs(destination)
+        shutil.copystat(source, destination)
```

Comment:
This is what we are trying to avoid -- copying the directory permissions. We don't need to copy the directory permissions, only the files over. The issue with copytree is that it copies the permissions and times of **directories** are copied with [copystat()](https://docs.python.org/3/library/shutil.html#shutil.copystat), individual files are copied using [copy2()](https://docs.python.org/3/library/shutil.html#shutil.copy2).

Reply:
Ah is the main issue that you replicated is that if the destination directory already exists (such as if writing directly to the PVC path mounted and not an inner directory) then when copying over the final checkpoint the script failed? But then if the directory didn't already exists then the script would create the directory and that succeeds?